string-methods/challenges-05.test.js

```diff
@@ -0,0 +1,358 @@
+------------------------------------------------------------------------------------------------ */
+
+const removeEvenValues = (arr) => {
+  // Solution code here...
+  let array = [...arr];
+
+  arr.splice(0, arr.length);
+
+  array.forEach(a => {
+    if (a % 2 !== 0) {
+      arr.push(a);
+    }
+  });
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 7
+
+Write a function named removeLastCharacters that takes in a string and a number. The numberOfCharacters argument determines how many characters will be removed from the end of the string. Return the resulting string.
+
+If the numberOfCharacters argument is greater than the length of the input string, the function should return an empty string.
+
+If the numberOfCharacters argument input is a negative number, the function should return the input string without any changes.
+
+For example:
+removeLastCharacters('Gregor', 2) returns 'Greg'
+removeLastCharacters('Gregor', -2) returns 'Gregor'
+removeLastCharacters('Gregor', 9) returns ''
+------------------------------------------------------------------------------------------------ */
+
+const removeLastCharacters = (str, numberOfCharacters) => {
+  // Solution code here...
+  // (numberOfCharacters > str) : return '' ? 
+  let stringLength = str.length;
+
+  if (stringLength < numberOfCharacters) {
+    return '';
+  } else if (numberOfCharacters < 0) {
+    return str;
+  } else {
+    let stringArray = str.split('');
+    stringArray.splice(stringLength - numberOfCharacters, stringLength);
+    let string = stringArray.join('');
+    return string;
+  }
+};
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 8 - Stretch Goal
+
+Write a function named totalSumCSV that, given a string of comma-separated values (CSV) as input. (e.g. "1,2,3"), returns the total sum of the numeric values (e.g. 6).
+------------------------------------------------------------------------------------------------ */
+
+const totalSumCSV = (str) => {
+  let total = 0;
+  // Solution code here...
+  let numberArray = [];
+  let stringArray = str.split(',');
+
+  stringArray.forEach(s => numberArray.push(parseInt(s)));
+
+  console.log(numberArray);
+
+  for (let i = 0; i < numberArray.length; i ++){
+    total += numberArray[i];
+  }
+
+  return total;
+};
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 9 - Stretch Goal
+
+Write a function named removeVowels that takes in a string and returns a new string where all the vowels of the original string have been removed.
+
+For example, removeVowels('gregor') returns 'grgr'.
+------------------------------------------------------------------------------------------------ */
+
+const removeVowels = (str) => {
+  // Solution code here...
+  return str.replace(/[AaEeIiOoUu]/g,'').split('').join('');
```

Comment:
Since you are using replace, you could have just done:
return str.replace(/[AaEeIiOoUu]/g,'');
The .split() and .join() are extra